### PR TITLE
Whitelist fields

### PR DESCRIPTION
**Pull request by mkotsollaris**

### CODE REVIEW

File: src/actions/index.js [hunk 1]
```diff
@@ -98,9 +98,9 @@ const generateConsentSchema = (consents) => {
   return schema;
 };
 
-const validateCustomFields = (customFields) => {
-  if (customFields) {
-    Object.keys(customFields).forEach((key) => {
+const validateAdditionalFields = (additionalFields) => {
+  if (additionalFields) {
```
Comment (marat-asadurian) [inline src/actions/index.js]:
```suggestion
  if (
    !additionalFields || 
    !Array.isArray(additionalFields) || 
    !additionalFields.find(key => reservedFormDataKeys.includes(key)) {
      throw...
    }
```

File: src/actions/index.js [hunk 2]
```diff
@@ -121,11 +121,51 @@ const parseTruliooFields = (formData) => {
   return truliooFields;
 };
 
-export const getFields = (countryCode, customFields) => async (dispatch) => {
+/**
+ * Returns the json-schema friendly whitelisted fields
+ *
+ * @param {TruliooFields} fields
```
Comment (marat-asadurian) [inline src/actions/index.js]:
```suggestion
 * @param {Trulioo fields} fields
```

File: src/actions/index.js [hunk 3]
```diff
@@ -121,11 +121,51 @@ const parseTruliooFields = (formData) => {
...
+ * @param {white-listed Trulioo Fields} whiteListedTruliooFields
+ * @param {resulting object} whiteListedComputedFields
+ */
+const getWhiteListedFieldsOnly = (fields, whiteListedTruliooFields, whiteListedComputedFields) => {
+  Object.keys(whiteListedTruliooFields).forEach((key) => {
+    const keyExists = Object.prototype.hasOwnProperty.call(fields, key);
+    // key is contained in fields
+    if (keyExists) {
+      const hasDefinedChildren = Object.keys(whiteListedTruliooFields[key]).length > 0;
+      if (hasDefinedChildren) {
```
Comment (marat-asadurian) [inline src/actions/index.js]: invert if and do early return

File: src/actions/index.js (hunk 1 above)
Comment (mkotsollaris) [reply src/actions/index.js]: 👍

File: TRULIOO_FIELDS.md [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+        "MonthOfBirth": 1,
+        "DayOfBirth": 1,
+        "ISOLatin1Name": "string",
+        "SecondSurname": "string",
+        "FirstSurName": "string",
+        "MiddleName": "string",
+        "FirstGivenName": "string"
+    },
+    "Location": {
+        "AdditionalFields": { "Address1": "string" },
```
Comment (kateika) [inline TRULIOO_FIELDS.md]: do we support Address2 line as well?

File: TRULIOO_FIELDS.md [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+        "POBox": "string",
+        "PostalCode": "string",
+        "Country": "string",
+        "StateProvinceCode": "string",
+        "County": "string",
```
Comment (kateika) [inline TRULIOO_FIELDS.md]: Country?

File: examples/src/index.jsx [hunk 6]
```diff
@@ -42,12 +42,24 @@ const customFields = {
...
+const whiteListedTruliooFields = {
+  properties: {
+    PersonInfo: {
+      properties: {
+        FirstGivenName: {
+        },
+      },
+    },
+  },
+};
```
Comment (kateika) [inline examples/src/index.jsx]: it's either } is missing or something with indents

File: src/tests/unit/EmbedID.test.jsx [hunk 7]
```diff
@@ -83,12 +85,46 @@ describe('EmbedID events function properly', () => {
...
       countries: {
-        'ui:title': 'Testing Countries Element',
-        'ui:description': 'Testing Description',
+        'ui:title': 'Please select your country of residence:',
+        'ui:description': 'Country Selection',
       },
     };
     const { container } = render(<EmbedID uiSchema={uiSchema} />);
-    const countriesLabel = getByText(container, /Testing Countries Element/);
+    const countriesLabel = getByText(container, /Please select your country/);
```
Comment (kateika) [inline src/tests/unit/EmbedID.test.jsx]: why we put colon in ' Please select your country of residence: ' but don't put here?

File: src/tests/unit/EmbedID.test.jsx [hunk 8]
```diff
@@ -83,12 +85,46 @@ describe('EmbedID events function properly', () => {
...
+    const whiteListedTruliooFields = {
+      properties: {
+        PersonInfo: {
+          properties: {
+            FirstGivenName: {
+            },
+          },
+        },
+      },
+    };
```
Comment (kateika) [inline src/tests/unit/EmbedID.test.jsx]: same here about indents and }

File: TRULIOO_FIELDS.md (hunk 5 above)
Comment (mkotsollaris) [reply TRULIOO_FIELDS.md]: yes this is legit; we do support counties as well

File: src/tests/unit/EmbedID.test.jsx (hunk 7 above)
Comment (mkotsollaris) [reply src/tests/unit/EmbedID.test.jsx]: Good suggestion!

**Pull request merged**